client.go

```diff
@@ -115,32 +116,37 @@ func (c *Client) Call(ctx context.Context, method, path string, idempotencyKey s
 		}
 	}()
 
-	switch {
-	case resp.StatusCode >= http.StatusInternalServerError, resp.StatusCode == http.StatusTooManyRequests:
-		return resp.StatusCode, ServerError{StatusCode: resp.StatusCode}
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return resp.StatusCode, errors.Wrap(err, "failed to read response body")
+	}
 
-	case resp.StatusCode == http.StatusUnprocessableEntity:
-		respBody, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return resp.StatusCode, errors.Wrap(err, "failed to read response body")
-		}
-		var validationError ValidationError
-		if err := json.Unmarshal(respBody, &validationError); err != nil {
-			return resp.StatusCode, errors.Wrapf(err, "failed to unmarshal response error with status %d: %s", resp.StatusCode, string(respBody))
+	if resp.StatusCode >= http.StatusBadRequest {
+		switch {
+		case resp.StatusCode == http.StatusUnauthorized, resp.StatusCode == http.StatusTooManyRequests:
+			return resp.StatusCode, ServerError{
+				StatusCode: resp.StatusCode,
+			}
+		case resp.StatusCode >= http.StatusInternalServerError, resp.StatusCode == http.StatusTooManyRequests, resp.StatusCode == http.StatusUnprocessableEntity:
```

Comment:
`http.StatusTooManyRequests` is checked twice, please see previous case.

Reply:
Fixed, thanks!